include/deal.II/lac/petsc_precondition.h

```diff
@@ -62,7 +62,7 @@ namespace PETScWrappers
     /**
      * Constructor.
      */
-    PreconditionBase();
+    PreconditionBase(const MPI_Comm &mpi_communicator = MPI_COMM_NULL);
```

Comment:
I think we should deprecate the old constructor then.

Reply:
Yes, and it should probably be made explicit?

Reply:
I don't think this is really necessary. This is, for all intents and purposes an internal class. It is of course conceivable that some user is wrapping another PETSc preconditioner, but I think it is not particularly likely. If you insist, I can put the old constructor back in, of course.